syntax = "proto3";
// for Golang
package hello;

option go_package = "github.com/avinassh/grpc-errors/go/hello";

service HelloService {
    // This thing just says Hello to anyone
    // SayHello('Euler') -> Hello, Euler!
    rpc SayHello(HelloReq) returns (HelloResp) {};
    // Strict Version responds only to requests which have `Name` length
    // less than 10 characters
    rpc SayHelloStrict(HelloReq) returns (HelloResp) {};
}

message HelloReq {
    string Name = 1;
}

message HelloResp {
    string Result = 1;
}
